### PR TITLE
Add Guidepost Sports quest

### LINKED ISSUE #467: New Quest: For which sport is this guidepost intended? (information=guidepost/route_marker)
Reference: https://github.com/streetcomplete/StreetComplete/issues/4758
And: https://github.com/streetcomplete/StreetComplete/issues/5208

### General
Affected tag(s) to be modified/added: The common and established tags for sports (like `hiking`, `bicycle`, `ski`, ...
Question asked: **For which sport is this guidepost intended?**
The goal is to determine the particular sport(s) for a guidepost or routemarker.

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)):
- [x] 🚧 To be added tag is established and has a useful purpose
- [ ] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by any pedestrian from the outside but a survey is necessary
- [x] 💤 Not an overwhelming percentage of quests have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of map data (Worth the effort)

Data from Jan 18:
Number of matched nodes in Austria: 14310
Number of matched nodes in Germany: 37408
Number of matched nodes in Switzerland: 7956
Number of matched nodes in United States: 10388

Overpass-Link: https://overpass-turbo.eu/s/1qhA

### Ideas for implementation

**Element selection:** 
```
nodes and 
["tourism"="information"] and
["information"="guidepost"] or ["information"="route_marker"] and
[!"hiking|bicycle|mtb|climbing|horse|nording_walking|ski|inline_skates|running"] and
[!"disused"][!"guidepost"]
```

There are certainly more sports to filter. Here are the ones that are known to me so far. Please comment if I've missed one!

**Concerns:** 

- It often happens that there are several signs for different sports hanging on a pole. Since the ref=* is usually also different, I have preferred to enter the nodes separately. This would not be possible in the Quest. A multiple selection would then have to be stored using a semicolon.
- There are different ways to tag the sport. Either with "hiking=yes" or "guidepost=hiking" or "guidepost:hiking=yes". Based on an evaluation of taginfo, the first one in the list is the most used.
- There is no "no" answer. But from my point of view, each signpost has a specific sport (at least from my regional experience). If the user can not say it it definitely needs a "can't say" answer option if it is not answerable to the user.
- If there is no specific sport displayed on the guidepost, there is `guidepost=simple` in the wiki: https://wiki.openstreetmap.org/wiki/Tag%3Aguidepost%3Dsimple

**Proposed UI:**

- Simple image selection or icon selection as in the recycling quest.
- You should be able to specify "This is a routemarker" as an alternative answer. This answer is more for experienced users, but it makes the usability easier and the effort is manageable.

** Possible Quest Filter
```
nodes with
          tourism = information
          and (information ~ guidepost|route_marker|map)
          and (!guidepost and !disused
          and (
                    !hiking or !bicycle or !mtb or !climbing or !horse or !nording_walking or! ski or !inline_skates or !running
          )
```

### PR DESCRIPTION
**Pull request by mcliquid**
Fixes #467

### CODE REVIEW

**Comment by Helium314** [comment]
Considering https://github.com/streetcomplete/StreetComplete/issues/5412, this PR should be done before the mentioned changes.
Do you want to continue here, or do you need anything from me?
As far as I understand from #467 you already have found some icons, so there should not be much work left.

File: app/src/main/java/de/westnordost/streetcomplete/quests/guidepost_sport/AddGuidepostSports.kt [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    private val filter by lazy { """
+        nodes with
+          tourism = information
+          and information ~ guidepost|route_marker
+          and !hiking and !bicycle and !mtb and !climbing and !horse and !nordic_walking and !ski and !inline_skates and !running
+          and !disused
+          and !guidepost
+    """.toElementFilterExpression() }
+
```
Comment (qugebert) [inline app/src/main/java/de/westnordost/streetcomplete/quests/guidepost_sport/AddGuidepostSports.kt:24]: Maybe for maps too?
At least in my region, information=map is often dedicated to bicycle or hiking routes.

Comment (qugebert) [reply app/src/main/java/de/westnordost/streetcomplete/quests/guidepost_sport/AddGuidepostSports.kt:24]: Sure, this is ok too

File: app/src/main/java/de/westnordost/streetcomplete/quests/guidepost_sport/AddGuidepostSportsForm.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package de.westnordost.streetcomplete.quests.guidepost_sport
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.quests.AImageListQuestForm
+import de.westnordost.streetcomplete.view.image_select.ImageListPickerDialog
+import de.westnordost.streetcomplete.view.image_select.Item
+
+class AddGuidepostSportsForm : AImageListQuestForm<List<GuidepostSport>, GuidepostSportsAnswer>() {
+
+    override val descriptionResId = R.string.quest_recycling_materials_note
```
Comment (qugebert) [inline app/src/main/java/de/westnordost/streetcomplete/quests/guidepost_sport/AddGuidepostSportsForm.kt:10]: I guess this needs still to be changed. See your screenshot
https://github.com/Helium314/SCEE/assets/3351668/f0e931d1-7572-46a6-a878-025acfce6764
"If anything not listed can be recycled here [...] "

**Comment by Helium314** [comment]
You will need to adjust the layout xml and set the new layout in the quest form in `onCreate`, e.g. `imageSelector.cellLayoutId = R.layout.cell_icon_select_with_label_below`
But note that it should also look ok with (nearly) black background

**Comment by Helium314** [comment]
I noticed the recycling quest does have a text color suitable for the chosen theme, so there should be some `cellLayoutId` you can use.

**Comment by Helium314** [comment]
Do you still want to add the `guidepost=simple` other answer?

File: app/src/main/java/de/westnordost/streetcomplete/quests/guidepost_sport/AddGuidepostSports.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    override fun createForm() = AddGuidepostSportsForm()
+
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("nodes with tourism = information and information ~ guidepost|route_marker")
+
+    override fun applyAnswerTo(answer: GuidepostSportsAnswer, tags: Tags, geometry: ElementGeometry, timestampEdited: Long) {
+        if (answer is IsSimpleGuidepost) {
+            applySimpleGuidepostAnswer(tags)
+        } else if (answer is SelectedGuidepostSports) {}
```
Comment (Helium314) [inline app/src/main/java/de/westnordost/streetcomplete/quests/guidepost_sport/AddGuidepostSports.kt:47]: You want to add `<sport>=yes`, right?
This should be done with `answer.selectedSports.forEach { tags[it.key] = "yes" }`

**Comment by Helium314** [comment]
The answers should use actual sport, having _Bicycle_, _Horse_ and _Ski_ in there feels wrong.
The quest looks like it could actually be an `OsmFilterQuestType` instead of `OsmElementQuestType`.

Otherwise it looks good!

**Pull request merged**